clients/fides-js/src/components/tcf/TcfOverlay.tsx

```diff
@@ -78,14 +83,24 @@ const getEnabledIds = (modelList: TcfModels) => {
     .map((model) => `${model.id}`);
 };
 
-export interface EnabledIds {
-  purposes: string[];
-  specialPurposes: string[];
-  features: string[];
-  specialFeatures: string[];
-  vendors: string[];
-  systems: string[];
-}
+const getVendorEnabledIds = (
+  modelList: TCFVendorRecord[] | undefined,
+  legalBasis: LegalBasisForProcessingEnum
+) => {
+  if (!modelList) {
+    return [];
+  }
+  const records = vendorRecordsWithLegalBasis(modelList, legalBasis);
+  if (legalBasis === LegalBasisForProcessingEnum.LEGITIMATE_INTERESTS) {
+    // TODO: the backend should eventually return legint fields with a default preference of OPT_IN
```

Comment:
do we have a ticket number this is coming in?

Reply:
I think that'd fall under this one: https://github.com/ethyca/fides/issues/4141 I'll add it to the comment 👍